TESTING.md

```diff
@@ -0,0 +1,182 @@
+
+```
+14:24:00.021:  Invoking handler gpii.preferencesServer.get.handler for route /preferences/:gpiiKey with expectedGrade kettle.request.http
+14:24:00.025:  Kettle server allocated request object with type gpii.preferencesServer.get.handler
+14:24:00.025:  gpii.preferencesServer.getPreferences called - fetching preferences for the GPII key testUser1
+14:24:00.025:  DataSource Issuing HTTP request with options {
+    "port": "5984",
+    "method": "GET",
+    "headers": {
+
+    },
+    "protocol": "http:",
+    "auth": "9af19d4cc412590b61e2440ce29eff2c:72e93ea87d01b2685950c01354e9ec70",
+    "host": "couchdb-svc-couchdb.gpii.svc.cluster.local:5984",
+    "hostname": "couchdb-svc-couchdb.gpii.svc.cluster.local",
+    "path": "/gpii/_design/views/_view/findPrefsSafeByGpiiKey?key=%22testUser1%22&include_docs=true",
+    "termMap": {
+        "gpiiKey": "%gpiiKey",
+        "baseUrl": "noencode:%baseUrl",
+        "port": "%port",
+        "dbName": "%dbName"
+    },
+    "url": "%baseUrl:%port/%dbName/_design/views/_view/findPrefsSafeByGpiiKey?key=%22%gpiiKey%22&include_docs=true",
+    "directModel": {
+        "baseUrl": "http://9af19d4cc412590b61e2440ce29eff2c:72e93ea87d01b2685950c01354e9ec70@couchdb-svc-couchdb.gpii.svc.cluster.local",
+        "port": "5984",
+        "dbName": "gpii",
+        "gpiiKey": "testUser1"
+    },
+    "operation": "get",
+    "reverse": false,
+    "writeMethod": "PUT",
+    "notFoundIsEmpty": true
+}
+14:24:00.097:  Preferences Server, getPreferences(), returning preferences: {
+    "contexts": {
+        "gpii-default": {
+            "name": "Default preferences",
+            "preferences": {
+                "http://registry.gpii.net/common/setting1": 12,
+                "http://registry.gpii.net/common/setting2": "white",
+                "http://registry.gpii.net/common/setting3": "black",
+                "http://registry.gpii.net/common/setting4": [
+                    "Comic Sans"
+                ],
+                "http://registry.gpii.net/common/setting5": "sans serif",
+                "http://registry.gpii.net/common/setting6": false
+            }
+        }
+    },
+    "name": undefined
+}
+```
+
+The above functionality is a limited test suite that needs to be expanded to be more complete. Work is being tracked in [GPII-3333](https://issues.gpii.net/browse/GPII-3333) for that.
+
+## Testing login/logout and basic settings
+
+1. Obtain a copy of the gpii/gpii-app repository:
+
+```bash
+$ git clone https://github.com/GPII/gpii-app.git
+
+```
+
+2. Spin up the vagrant box:
+
+```bash
+$ cd gpii-app
+
+$ vagrant up
+```
+
+3. When the machine is up and running, open a command prompt and run the application:
+
+```
+$ cd c:\vagrant
+
+$ set GPII_CLOUD_URL=http://flowmanager.<CLUSTER_NAME>
+
+$ npm start
+
+...
+
+```
+
+4. You can open morphic now on the taskbar. The icon will of a gear. Update of the preferences should now register against the specified cluster.
```

Comment:
Maybe "Preferences changed in Morphic should save to the specified cluster."?

Reply:
Yeah, I think that it would be great to have exactly the steps required to produce a "save" action in the cloud and how to check the flow of data going back and forth from one point to the other.